File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/search/TypoTolerance.java
package com.algolia.model.search;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Controls whether typo tolerance is enabled and how it is applied. */
@JsonAdapter(TypoTolerance.Adapter.class)
public enum TypoTolerance {
  TRUE("true"),

  FALSE("false"),

  MIN("min"),

  STRICT("strict");

  private final String value;

  TypoTolerance(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static TypoTolerance fromValue(String value) {
    for (TypoTolerance b : TypoTolerance.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<TypoTolerance> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final TypoTolerance enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public TypoTolerance read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return TypoTolerance.fromValue(value);
    }
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/TypoTolerance.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * TypoTolerance Class Doc Comment
 *
 * @category Class
 * @description Controls whether typo tolerance is enabled and how it is applied.
 *
 * @package Algolia\AlgoliaSearch
 */
class TypoTolerance
{
    /**
     * Possible values of this enum
     */
    const TRUE = 'true';

    const FALSE = 'false';

    const MIN = 'min';

    const STRICT = 'strict';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::TRUE,
            self::FALSE,
            self::MIN,
            self::STRICT,
        ];
    }
}

